File: app/src/main/java/com/theroyalsoft/telefarmer/model/ProxyVideoSink.java
package com.theroyalsoft.telefarmer.model;

import org.webrtc.VideoFrame;
import org.webrtc.VideoSink;

import timber.log.Timber;

/**
 * Created by Pritom Dutta on 5/8/23.
 */
public class ProxyVideoSink implements VideoSink {
    private VideoSink mTarget;

    @Override
    synchronized public void onFrame(VideoFrame frame) {
        if (mTarget == null) {
            Timber.d("Dropping frame in proxy because target is null.");
            return;
        }
        mTarget.onFrame(frame);
    }

    public   synchronized void setTarget(VideoSink target) {
        this.mTarget = target;
    }
}

File: callingWebrtc/src/main/java/bio/medico/patient/callingWebrtc/model/ProxyVideoSink.kt
package bio.medico.patient.callingWebrtc.model

import org.webrtc.VideoFrame
import org.webrtc.VideoSink
import timber.log.Timber

class ProxyVideoSink : VideoSink {
    private lateinit var mTarget: VideoSink
    @Synchronized
    override fun onFrame(frame: VideoFrame) {
        if (mTarget == null) {
            Timber.d("Dropping frame in proxy because target is null.")
            return
        }
        mTarget.onFrame(frame)
    }

    @Synchronized
    fun setTarget(target: VideoSink) {
        mTarget = target
    }
}